modules/collect/src/main/java/com/opengamma/strata/collect/Guavate.java

```diff
@@ -588,4 +621,64 @@ public R next() {
   public static <K, V> Collector<Pair<K, V>, ?, ImmutableMap<K, V>> pairsToImmutableMap() {
     return toImmutableMap(Pair::getFirst, Pair::getSecond);
   }
+
+  //--------------------------------------------------------------------------------------------------
+
+  /**
+   * Helper method which puts a value into a mutable map.
+   * <p>
+   * If the map already contains a mapping for the key the merge function is applied to the existing value and
+   * the new value, and the return value is inserted.
+   *
+   * @param map  the map
+   * @param key  the key
+   * @param val  the value
+   * @param mergeFn  function applied to the existing and new values if the map contains the key
+   * @param <K>  the key type
+   * @param <V>  the value type
+   */
+  private static <K, V> void putWithMerge(
+      Map<K, V> map,
+      K key,
+      V val,
+      BiFunction<? super V, ? super V, ? extends V> mergeFn) {
+
+    V existingVal = map.get(key);
+
+    if (existingVal == null) {
+      map.put(key, val);
+    } else {
+      map.put(key, mergeFn.apply(existingVal, val));
+    }
+  }
+
+  /**
+   * Helper method to merge two mutable maps by inserting all values from {@code map2} into {@code map1}.
+   * <p>
+   * If {@code map1} already contains a mapping for a key the merge function is applied to the existing value and
+   * the new value, and the return value is inserted.
+   *
+   * @param map1  the map into which values are copied
+   * @param map2  the map from which values are copied
+   * @param mergeFn  function applied to the existing and new values if the map contains the key
+   * @param <K>  the key type
+   * @param <V>  the value type
+   * @return {@code map1} with the values from {@code map2} inserted
+   */
+  private static <K, V> Map<K, V> mergeMaps(
```

Comment:
I'm at the mercy of the signature of `Collector.of` which required a `BinaryOperator`.